pallets/parachain-staking/src/types.rs

```diff
@@ -458,6 +458,37 @@ impl<
 		});
 		Ok(())
 	}
+
+	pub fn bond_less<T: Config>(&mut self, who: T::AccountId, amount: Balance)
+	where
+		BalanceOf<T>: From<Balance>,
+	{
+		let new_total_staked = <Total<T>>::get().saturating_sub(amount.into());
+		<Total<T>>::put(new_total_staked);
+		self.bond = self.bond.saturating_sub(amount);
+		if self.bond.is_zero() {
+			T::Currency::remove_lock(COLLATOR_LOCK_ID, &who);
+		} else {
+			T::Currency::set_lock(
+				COLLATOR_LOCK_ID,
+				&who,
+				self.bond.into(),
+				WithdrawReasons::all(),
+			);
+		}
```

Comment:
Is this case even possible anymore? My understanding is that the entire idea of `COLLATOR_LOCK_ID` goes away and after this PR (and its migration) there should never be such a lock again, is that right?

If so:
1. what happens when someone has a scheduled `bond_less` in progress? I would make sure it's a no-op in the most resilient way possible.
1. we should prevent any new scheduled collator bonds from being scheduled
1. we should plan to clean this up in the next release by removing code and making sure all on-chain scheduled bond-less are removed

Reply:
There seems to be a misunderstanding: We remove the bonds of the orbiter collators only. The regular collators are not effected. Can you verify that please @crystalin @noandrea ?

Reply:
Oh, I see. I missed that in a couple of places in my earlier review, looks good to me now.

Reply:
Yes, I confirm it is only for the "community/system collators" (which currently are identified as the orbiter pools accounts)